internal/services/loadbalancer/rule_resource.go

```diff
@@ -245,6 +249,10 @@ func resourceArmLoadBalancerRuleDelete(d *pluginsdk.ResourceData, meta interface
 		return fmt.Errorf("failed to retrieve Load Balancer %q (resource group %q) for Rule %q: %+v", id.LoadBalancerName, id.ResourceGroup, id.Name, err)
 	}
 
+	if loadBalancer.LoadBalancerPropertiesFormat == nil {
+		return fmt.Errorf("retrieving Load Balancer %q (resource group %q) for Rule %q: `properties` was nil", id.LoadBalancerName, id.ResourceGroup, id.Name)
+	}
+
```

Comment:
same here, we should be wrapping the use of the property in a != nil rather then erroring out